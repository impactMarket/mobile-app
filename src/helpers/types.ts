import { ContractKit } from '@celo/contractkit';

export const STORAGE_USER_ADDRESS = '@celoinfo:address';
export const STORAGE_USER_PHONE_NUMBER = '@celoinfo:phonenumber';
export const STORAGE_USER_FIRST_TIME = '@status:firstime';
export const STORAGE_USER_AUTH_TOKEN = '@user:authtoken';
export const INIT_USER = 'INIT_USER';
export const SET_USER_CELO_INFO = 'SET_USER_CELO_INFO';
export const SET_USER_INFO = 'SET_USER_INFO';
export const SET_USER_EXCHANGE_RATE = 'SET_USER_EXCHANGE_RATE';
export const SET_USER_WALLET_BALANCE = 'SET_USER_WALLET_BALANCE';
export const SET_CELO_KIT = 'SET_CELO_KIT';
export const SET_COMMUNITY_CONTRACT = 'SET_COMMUNITY_CONTRACT';
export const SET_COMMUNITY = 'SET_COMMUNITY';
export const SET_USER_IS_BENEFICIARY = 'SET_USER_IS_BENEFICIARY';
export const SET_USER_IS_COMMUNITY_MANAGER = 'SET_USER_IS_COMMUNITY_MANAGER';
export const RESET_USER_APP = 'RESET_USER_APP';
export const RESET_NETWORK_APP = 'RESET_NETWORK_APP';
export const SET_PUSH_NOTIFICATION_TOKEN = 'SET_PUSH_NOTIFICATION_TOKEN';
export const SET_AUTH_TOKEN = 'SET_AUTH_TOKEN';
export const SET_APP_PYMENT_TO_ACTION = 'SET_APP_PYMENT_TO_ACTION';
export const SET_USER_LANGUAGE = 'SET_USER_LANGUAGE';
export const SET_EXCHANGE_RATES = 'SET_EXCHANGE_RATES';
export const CONSENT_ANALYTICS = 'CONSENT_ANALYTICS';

// state
export interface IUserCeloInfo {
    // verify if address is undefined to determine if user is logged in
    address: string;
    phoneNumber: string;
    balance: string;
}

// TODO: same as IUser. Rmove one.
export interface IUserInfo {
    name: string;
    currency: string;
    exchangeRate: number;
    avatar: string;
    language: string;
}

export interface IUserCommunityInfo {
    isBeneficiary: boolean;
    isManager: boolean;
}

export interface IUserState {
    celoInfo: IUserCeloInfo;
    user: IUserInfo;
    community: IUserCommunityInfo;
}

export interface IContractsState {
    communityContract: any;
}

export interface INetworkState {
    community: ICommunityInfo;
    contracts: IContractsState;
}

export interface IAuthState {
    pushNotificationsToken: string;
    authToken: string;
}

export interface IAppState {
    kit: ContractKit;
    exchangeRates: any;
    paymentToAddress: string;
}

// same as ICombinedState
export interface IRootState {
    user: IUserState;
    network: INetworkState;
    auth: IAuthState;
    app: IAppState;
}

// action
interface UserCeloInfoAction {
    type: typeof SET_USER_CELO_INFO;
    payload: IUserCeloInfo;
}

interface UserInfoAction {
    type: typeof SET_USER_INFO;
    payload: IUserInfo;
}

interface UserExchangeRateAction {
    type: typeof SET_USER_EXCHANGE_RATE;
    payload: number;
}

interface InitUserAction {
    type: typeof INIT_USER;
    payload: IInitUser;
}

interface UserSetBalanceAction {
    type: typeof SET_USER_WALLET_BALANCE;
    payload: string;
}

interface UserSetIsBeneficiaryAction {
    type: typeof SET_USER_IS_BENEFICIARY;
    payload: boolean;
}

interface UserSetIsCommunityManagerAction {
    type: typeof SET_USER_IS_COMMUNITY_MANAGER;
    payload: boolean;
}

interface CeloKitAction {
    type: typeof SET_CELO_KIT;
    payload: ContractKit;
}

interface SetCommunityContractAction {
    type: typeof SET_COMMUNITY_CONTRACT;
    payload: any;
}

interface SetCommunityAction {
    type: typeof SET_COMMUNITY;
    payload: ICommunityInfo;
}

interface ResetUserAction {
    type: typeof RESET_USER_APP;
    payload: any;
}

interface ResetNetworkAction {
    type: typeof RESET_NETWORK_APP;
    payload: any;
}

interface SetTokenPushNotificationsAction {
    type: typeof SET_PUSH_NOTIFICATION_TOKEN;
    payload: string;
}

interface SetAuthTokenAction {
    type: typeof SET_AUTH_TOKEN;
    payload: string;
}
interface UserLanguageAction {
    type: typeof SET_USER_LANGUAGE;
    payload: string;
}

interface SetAppPaymentToAction {
    type: typeof SET_APP_PYMENT_TO_ACTION;
    payload: string;
}

interface SetAppEchangeRatesAction {
    type: typeof SET_EXCHANGE_RATES;
    payload: any;
}

export type UserActionTypes =
    | UserCeloInfoAction
    | UserSetBalanceAction
    | UserSetIsBeneficiaryAction
    | UserSetIsCommunityManagerAction
    | ResetUserAction
    | UserInfoAction
    | UserLanguageAction
    | UserExchangeRateAction
    | InitUserAction;
export type NetworkActionTypes =
    | SetCommunityContractAction
    | SetCommunityAction
    | ResetNetworkAction;
export type AuthActionTypes =
    | SetTokenPushNotificationsAction
    | SetAuthTokenAction;
export type AppActionTypes =
    | CeloKitAction
    | SetAppPaymentToAction
    | SetAppEchangeRatesAction;

export interface IStoreCombinedState {
    user: IUserState;
    network: INetworkState;
    auth: IAuthState;
    app: IAppState;
}

export type IStoreCombinedActionsTypes =
    | UserActionTypes
    | NetworkActionTypes
    | AuthActionTypes
    | AppActionTypes;

export interface ITransaction {
    tx: string;
    from: string;
    contractAddress: string;
    event: string;
    values: any;
}

export interface ITabBarIconProps {
    focused: boolean;
    color: string;
    size: number;
}

// **API and app**

export interface ICommunity {
    publicId: string;
    requestByAddress: string;
    contractAddress: string;
    name: string;
    description: string;
    descriptionEn: string;
    language: string;
    currency: string;
    country: string;
    city: string;
    gps: {
        latitude: number;
        longitude: number;
    };
    visibility: string;
    email: string;
    coverImage: string;
    status: string;
    txCreationObj: ICommunityVars;
    createdAt: string;
    updatedAt: string;
}

export interface ICommunityInfo extends ICommunity {
    backers: string[];
    beneficiaries: {
        added: ICommunityInfoBeneficiary[];
        removed: ICommunityInfoBeneficiary[];
    };
    managers: string[];
    ssi: {
        dates: Date[];
        values: number[];
    };
    totalClaimed: string;
    totalRaised: string;
    vars: ICommunityVars;
}

export interface ICommunityVars {
    _claimAmount: string;
    _baseInterval: string;
    _incrementInterval: string;
    _maxClaim: string;
}

export interface ICommunityInfoBeneficiary {
    address: string;
    name: string;
    timestamp: number;
    claimed: string;
}

export interface IUser {
    address: string;
    username: string | null;
    currency: string;
    avatar: string;
    language: string;
}

export interface IUserWelcome {
    user: IUser; // TODO: remove in the future, as it's intended to be on memory
    exchangeRates: any; // TODO: this is not really an any
    isBeneficiary: boolean;
    isManager: boolean;
    community?: ICommunityInfo;
}

export interface IInitUser {
    user: IUser & IUserCeloInfo;
    exchangeRates: any;
    isBeneficiary: boolean;
    isManager: boolean;
    community?: ICommunityInfo;
}

export interface IUserWelcomeAuth extends IUserWelcome {
    token: string;
}

export interface IAddressAndName {
    address: string;
    name: string;
}

export interface IUserTxAPI {
    picture: string;
    counterParty: IAddressAndName;
    value: string;
    timestamp: number;
    fromUser: boolean;
}

/**
 * @deprecated
 */
export interface IRecentTxAPI {
    picture: string;
    from: IAddressAndName;
    value: string;
    timestamp: number;
}

/**
 * @deprecated
 */
export interface IPaymentsTxAPI {
    picture: string;
    to: IAddressAndName;
    value: string;
    timestamp: number;
}
