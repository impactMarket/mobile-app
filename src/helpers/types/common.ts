/**
 * @deprecated
 */
export interface IAddressAndName {
    address: string;
    name: string;
}

export interface ITabBarIconProps {
    focused: boolean;
    color: string;
    size: number;
}
