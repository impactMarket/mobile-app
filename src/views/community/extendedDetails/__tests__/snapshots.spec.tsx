import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { act, render } from '@testing-library/react-native';
import { CommunityAttributes, ManagerAttributes } from 'helpers/types/models';
import React from 'react';
import { Host } from 'react-native-portalize';
import * as reactRedux from 'react-redux';
import Api from 'services/api';

import CommunityExtendedDetailsScreen from '../';

jest.mock('components/community/Description');

const community: CommunityAttributes = {
    id: 1,
    publicId: '3f49b131-e097-4155-8f28-46c95590d42f',
    requestByAddress: '0xb10199414d158a264e25a5ec06b463c0cd8457bb',
    contractAddress: '0xd48466ffc8b6190d33fb6b27035a032658d392ee',
    name: 'Example Community',
    description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse quis interdum justo, nec sollicitudin lectus. Quisque malesuada metus lacinia facilisis pretium. Maecenas semper lacus id ante porttitor, non ultricies nunc suscipit. Vestibulum hendrerit eros vel mollis blandit. Nam tincidunt libero nunc, sed gravida tortor vulputate ut. Donec ullamcorper dui erat, vitae consectetur neque auctor sit amet. Aliquam a lacus congue, scelerisque orci vitae, accumsan tellus. Nunc ultrices convallis euismod.',
    city: 'Lauro de Freitas',
    country: 'BR',
    gps: { latitude: 1, longitude: 2 },
    email: 'eu@amigos.txt',
    visibility: 'public',
    status: 'valid',
    createdAt: new Date(),
    updatedAt: new Date(),
    currency: 'BRL',
    descriptionEn: null,
    language: 'pt',
    started: new Date(),
    coverMediaId: 260,
    contract: {
        claimAmount: '750000000000000000',
        maxClaim: '400000000000000000000',
        baseInterval: 86400,
        incrementInterval: 300,
        createdAt: new Date(),
        updatedAt: new Date(),
        communityId: 1,
    },
    state: {
        claimed: '61942500000000000000000',
        claims: 55405,
        beneficiaries: 656,
        raised: '62151145240504400207738',
        backers: 37,
        createdAt: new Date(),
        updatedAt: new Date(),
        removedBeneficiaries: 668,
        managers: 4,
        communityId: 1,
    },
};

/**
 * NOTE: we are testing the component individually, but need the header
 * "submit button", so the entire navigator can be faked.
 */
function WrappedCommunityDetailsScreen() {
    const Stack = createStackNavigator();
    return (
        <Host>
            <NavigationContainer>
                <Stack.Navigator>
                    <Stack.Screen
                        name="Home"
                        component={CommunityExtendedDetailsScreen}
                        initialParams={{
                            route: {
                                params: {
                                    communityId: 1,
                                },
                            },
                        }}
                    />
                </Stack.Navigator>
            </NavigationContainer>
        </Host>
    );
}

describe('extended details community [snapshot]', () => {
    const useSelectorMock = reactRedux.useSelector as jest.Mock<any, any>;

    const communityListManagersMock = jest.spyOn(Api.community, 'listManagers');

    beforeAll(() => {
        useSelectorMock.mockImplementation((callback) => {
            return callback({
                app: {
                    exchangeRates: { USD: 1 },
                },
                user: {
                    metadata: {
                        currency: 'USD',
                        avatar: 'something.jpg',
                    },
                    wallet: {
                        address: '0xd7632B7588DF8532C0aBA55586167C2a315Fd768',
                    },
                },
                communities: {
                    community,
                },
            });
        });

        communityListManagersMock.mockImplementation(() =>
            Promise.resolve([
                {
                    address: '0xb10199414d158a264e25a5ec06b463c0cd8457bb',
                    active: true,
                    communityId: '3f49b131-e097-4155-8f28-46c95590d42f',
                    id: 1,
                    user: {
                        address: '0xb10199414d158a264e25a5ec06b463c0cd8457bb',
                        children: 2,
                        currency: 'CELO',
                        gender: 'u',
                        language: 'pt',
                        lastLogin: new Date(1629972620000),
                        username: 'No Name',
                        pushNotificationToken: '',
                        year: 1990,
                        avatarMediaId: 0,
                        createdAt: new Date(1629972620000),
                        updatedAt: new Date(1629972620000),
                    },
                    createdAt: new Date(1629972620000),
                    updatedAt: new Date(1629972620000),
                },
                {
                    address: '0xc10199414d158a264e25a5ec06b463c0cd8457bb',
                    active: true,
                    communityId: '3f49b131-e097-4155-8f28-46c95590d42f',
                    id: 2,
                    user: {
                        address: '0xc10199414d158a264e25a5ec06b463c0cd8457bb',
                        children: 2,
                        currency: 'CELO',
                        gender: 'u',
                        language: 'pt',
                        lastLogin: new Date(1629972620000),
                        username: null,
                        pushNotificationToken: '',
                        year: 1990,
                        avatarMediaId: 0,
                        createdAt: new Date(1629972620000),
                        updatedAt: new Date(1629972620000),
                    },
                    createdAt: new Date(1629972620000),
                    updatedAt: new Date(1629972620000),
                },
                {
                    address: '0xd10199414d158a264e25a5ec06b463c0cd8457bb',
                    active: true,
                    communityId: '3f49b131-e097-4155-8f28-46c95590d42f',
                    id: 2,
                    createdAt: new Date(1629972620000),
                    updatedAt: new Date(1629972620000),
                },
            ] as ManagerAttributes[])
        );
    });

    it('renders correctly', async () => {
        const tree = render(<WrappedCommunityDetailsScreen />);
        const { queryAllByTestId } = tree;
        await act(async () => {});
        await act(async () => {
            queryAllByTestId('manager-box');
        });

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
