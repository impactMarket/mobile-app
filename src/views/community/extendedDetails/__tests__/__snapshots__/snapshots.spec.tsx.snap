// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extended details community [snapshot] renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <View>
      <Text
        style={
          Object {
            "color": "#333239",
            "fontFamily": "Manrope-Bold",
            "fontSize": 18,
            "lineHeight": 28,
          }
        }
      >
        Example Community
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={12}
          bbWidth={8}
          fill="none"
          focusable={false}
          height={12}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 12,
                "width": 8,
              },
            ]
          }
          vbHeight={12}
          vbWidth={8}
          width={8}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              clipRule={0}
              d="M0 4.535C0 2.412 1.794.685 4 .685s4 1.727 4 3.85c0 2.634-3.58 6.502-3.732 6.665a.37.37 0 01-.536 0C3.58 11.037 0 7.17 0 4.535zm1.988 0c0 1.068.902 1.937 2.012 1.937 1.11 0 2.012-.87 2.012-1.937 0-1.068-.902-1.937-2.012-1.937-1.11 0-2.012.869-2.012 1.937z"
              fill={4285760413}
              fillRule={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            Object {
              "color": "#73839D",
              "fontFamily": "Inter-Regular",
              "fontSize": 14,
              "lineHeight": 24,
              "marginLeft": 4,
            }
          }
        >
          Lauro de Freitas
          , 
          Brazil
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#333239",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "lineHeight": 24,
          "marginBottom": 28,
          "marginTop": 16,
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse quis interdum justo, nec sollicitudin lectus. Quisque malesuada metus lacinia facilisis pretium. Maecenas semper lacus id ante porttitor, non ultricies nunc suscipit. Vestibulum hendrerit eros vel mollis blandit. Nam tincidunt libero nunc, sed gravida tortor vulputate ut. Donec ullamcorper dui erat, vitae consectetur neque auctor sit amet. Aliquam a lacus congue, scelerisque orci vitae, accumsan tellus. Nunc ultrices convallis euismod.
    </Text>
    <View
      style={
        Object {
          "borderColor": "#E9EDF4",
          "borderRadius": 12,
          "borderWidth": 1,
          "marginVertical": 4,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#73839D",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Amount per Claim
      </Text>
      <Text
        style={
          Object {
            "color": "#333239",
            "fontFamily": "Inter-Bold",
            "fontSize": 16,
            "lineHeight": 22,
          }
        }
      >
        $0.75 / day
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#E9EDF4",
          "borderRadius": 12,
          "borderWidth": 1,
          "marginVertical": 4,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#73839D",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Amount
      </Text>
      <Text
        style={
          Object {
            "color": "#333239",
            "fontFamily": "Inter-Bold",
            "fontSize": 16,
            "lineHeight": 22,
          }
        }
      >
        $400
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#E9EDF4",
          "borderRadius": 12,
          "borderWidth": 1,
          "marginVertical": 4,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#73839D",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Incremental Frequency
      </Text>
      <Text
        style={
          Object {
            "color": "#333239",
            "fontFamily": "Inter-Bold",
            "fontSize": 16,
            "lineHeight": 22,
          }
        }
      >
        5 Minutes
      </Text>
    </View>
    <View
      style={
        Object {
          "borderColor": "#E9EDF4",
          "borderWidth": 1,
          "marginVertical": 32,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#333239",
          "fontFamily": "Inter-Bold",
          "fontSize": 20,
          "lineHeight": 20,
          "marginBottom": 8,
        }
      }
    >
      Managers
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#E9EDF4",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginVertical": 8,
          "padding": 16,
        }
      }
      testID="manager-box"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={42}
        bbWidth={42}
        fill="none"
        focusable={false}
        height={42}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "marginRight": 16,
            },
            Object {
              "flex": 0,
              "height": 42,
              "width": 42,
            },
          ]
        }
        vbHeight={80}
        vbWidth={81}
        width={42}
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGMask
            height={60}
            maskContentUnits={1}
            maskTransform={
              Array [
                1,
                0,
                -0,
                1,
                10,
                6,
              ]
            }
            maskUnits={0}
            name="prefix__a"
            width={61}
            x={10}
            y={6}
          >
            <RNSVGPath
              d="M69.5 35.958c0 15.9-12.977 28.8-29 28.8s-29-12.9-29-28.8 12.977-28.8 29-28.8 29 12.9 29 28.8z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth={2}
            />
          </RNSVGMask>
          <RNSVGGroup
            fill={4287274946}
            mask="prefix__a"
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M31.049 25.21c0-5.182 4.215-9.583 9.484-9.583 5.26 0 9.485 4.392 9.485 9.584 0 5.19-4.252 9.418-9.485 9.418-5.232 0-9.484-4.229-9.484-9.418zm3.992 0c0 3.014 2.467 5.46 5.492 5.46 3.026 0 5.493-2.446 5.493-5.46 0-3.041-2.498-5.624-5.493-5.624-2.994 0-5.492 2.583-5.492 5.625zM28.553 72.154v-.706l-.666.235a7 7 0 01-2.33.396c-3.856 0-6.989-3.116-6.989-6.939V50.263c0-6.557 5.372-11.898 11.981-11.898h19.97c6.609 0 11.981 5.34 11.981 11.898V64.12c0 3.823-3.132 6.94-6.989 6.94a7 7 0 01-2.33-.397l-.666-.235v24.633c0 3.822-3.132 6.938-6.988 6.938a6.986 6.986 0 01-4.66-1.772l-.333-.296-.332.296a6.986 6.986 0 01-4.66 1.772c-3.857 0-6.989-3.116-6.989-6.938V72.154zm6.989 25.886c.765 0 1.465-.287 1.996-.76v.173h1c0-1.09.89-1.98 1.996-1.98 1.106 0 1.996.89 1.996 1.98h1v-.172c.531.472 1.231.759 1.997.759a2.99 2.99 0 002.996-2.98V50.263c0-1.09.89-1.98 1.996-1.98 1.106 0 1.996.89 1.996 1.98V64.12a2.99 2.99 0 002.996 2.98 2.99 2.99 0 002.997-2.98V50.263c0-4.381-3.587-7.939-7.989-7.939h-19.97c-4.402 0-7.988 3.558-7.988 7.939V65.14a2.99 2.99 0 002.996 2.98 2.99 2.99 0 002.996-2.98V50.263c0-1.09.89-1.98 1.996-1.98 1.106 0 1.997.89 1.997 1.98V95.06a2.99 2.99 0 002.996 2.98z"
            />
          </RNSVGGroup>
          <RNSVGCircle
            cx={40.5}
            cy={40}
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            r={38.5}
            stroke={4287274946}
            strokeWidth={3}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333239",
              "fontFamily": "Inter-Bold",
              "fontSize": 13,
              "lineHeight": 22,
            }
          }
        >
          No Name
        </Text>
        <Text
          style={
            Object {
              "color": "#73839D",
              "fontFamily": "Inter-Regular",
              "fontSize": 12,
              "lineHeight": 20,
            }
          }
        >
          Manager since Ago 24, 2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#E9EDF4",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginVertical": 8,
          "padding": 16,
        }
      }
      testID="manager-box"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={42}
        bbWidth={42}
        fill="none"
        focusable={false}
        height={42}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "marginRight": 16,
            },
            Object {
              "flex": 0,
              "height": 42,
              "width": 42,
            },
          ]
        }
        vbHeight={80}
        vbWidth={81}
        width={42}
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGMask
            height={60}
            maskContentUnits={1}
            maskTransform={
              Array [
                1,
                0,
                -0,
                1,
                10,
                6,
              ]
            }
            maskUnits={0}
            name="prefix__a"
            width={61}
            x={10}
            y={6}
          >
            <RNSVGPath
              d="M69.5 35.958c0 15.9-12.977 28.8-29 28.8s-29-12.9-29-28.8 12.977-28.8 29-28.8 29 12.9 29 28.8z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth={2}
            />
          </RNSVGMask>
          <RNSVGGroup
            fill={4287274946}
            mask="prefix__a"
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M31.049 25.21c0-5.182 4.215-9.583 9.484-9.583 5.26 0 9.485 4.392 9.485 9.584 0 5.19-4.252 9.418-9.485 9.418-5.232 0-9.484-4.229-9.484-9.418zm3.992 0c0 3.014 2.467 5.46 5.492 5.46 3.026 0 5.493-2.446 5.493-5.46 0-3.041-2.498-5.624-5.493-5.624-2.994 0-5.492 2.583-5.492 5.625zM28.553 72.154v-.706l-.666.235a7 7 0 01-2.33.396c-3.856 0-6.989-3.116-6.989-6.939V50.263c0-6.557 5.372-11.898 11.981-11.898h19.97c6.609 0 11.981 5.34 11.981 11.898V64.12c0 3.823-3.132 6.94-6.989 6.94a7 7 0 01-2.33-.397l-.666-.235v24.633c0 3.822-3.132 6.938-6.988 6.938a6.986 6.986 0 01-4.66-1.772l-.333-.296-.332.296a6.986 6.986 0 01-4.66 1.772c-3.857 0-6.989-3.116-6.989-6.938V72.154zm6.989 25.886c.765 0 1.465-.287 1.996-.76v.173h1c0-1.09.89-1.98 1.996-1.98 1.106 0 1.996.89 1.996 1.98h1v-.172c.531.472 1.231.759 1.997.759a2.99 2.99 0 002.996-2.98V50.263c0-1.09.89-1.98 1.996-1.98 1.106 0 1.996.89 1.996 1.98V64.12a2.99 2.99 0 002.996 2.98 2.99 2.99 0 002.997-2.98V50.263c0-4.381-3.587-7.939-7.989-7.939h-19.97c-4.402 0-7.988 3.558-7.988 7.939V65.14a2.99 2.99 0 002.996 2.98 2.99 2.99 0 002.996-2.98V50.263c0-1.09.89-1.98 1.996-1.98 1.106 0 1.997.89 1.997 1.98V95.06a2.99 2.99 0 002.996 2.98z"
            />
          </RNSVGGroup>
          <RNSVGCircle
            cx={40.5}
            cy={40}
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            r={38.5}
            stroke={4287274946}
            strokeWidth={3}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333239",
              "fontFamily": "Inter-Bold",
              "fontSize": 13,
              "lineHeight": 22,
            }
          }
        >
          0xc10199414d...
        </Text>
        <Text
          style={
            Object {
              "color": "#73839D",
              "fontFamily": "Inter-Regular",
              "fontSize": 12,
              "lineHeight": 20,
            }
          }
        >
          Manager since Ago 24, 2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#E9EDF4",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginVertical": 8,
          "padding": 16,
        }
      }
      testID="manager-box"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={42}
        bbWidth={42}
        fill="none"
        focusable={false}
        height={42}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "marginRight": 16,
            },
            Object {
              "flex": 0,
              "height": 42,
              "width": 42,
            },
          ]
        }
        vbHeight={80}
        vbWidth={81}
        width={42}
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGMask
            height={60}
            maskContentUnits={1}
            maskTransform={
              Array [
                1,
                0,
                -0,
                1,
                10,
                6,
              ]
            }
            maskUnits={0}
            name="prefix__a"
            width={61}
            x={10}
            y={6}
          >
            <RNSVGPath
              d="M69.5 35.958c0 15.9-12.977 28.8-29 28.8s-29-12.9-29-28.8 12.977-28.8 29-28.8 29 12.9 29 28.8z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth={2}
            />
          </RNSVGMask>
          <RNSVGGroup
            fill={4287274946}
            mask="prefix__a"
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M31.049 25.21c0-5.182 4.215-9.583 9.484-9.583 5.26 0 9.485 4.392 9.485 9.584 0 5.19-4.252 9.418-9.485 9.418-5.232 0-9.484-4.229-9.484-9.418zm3.992 0c0 3.014 2.467 5.46 5.492 5.46 3.026 0 5.493-2.446 5.493-5.46 0-3.041-2.498-5.624-5.493-5.624-2.994 0-5.492 2.583-5.492 5.625zM28.553 72.154v-.706l-.666.235a7 7 0 01-2.33.396c-3.856 0-6.989-3.116-6.989-6.939V50.263c0-6.557 5.372-11.898 11.981-11.898h19.97c6.609 0 11.981 5.34 11.981 11.898V64.12c0 3.823-3.132 6.94-6.989 6.94a7 7 0 01-2.33-.397l-.666-.235v24.633c0 3.822-3.132 6.938-6.988 6.938a6.986 6.986 0 01-4.66-1.772l-.333-.296-.332.296a6.986 6.986 0 01-4.66 1.772c-3.857 0-6.989-3.116-6.989-6.938V72.154zm6.989 25.886c.765 0 1.465-.287 1.996-.76v.173h1c0-1.09.89-1.98 1.996-1.98 1.106 0 1.996.89 1.996 1.98h1v-.172c.531.472 1.231.759 1.997.759a2.99 2.99 0 002.996-2.98V50.263c0-1.09.89-1.98 1.996-1.98 1.106 0 1.996.89 1.996 1.98V64.12a2.99 2.99 0 002.996 2.98 2.99 2.99 0 002.997-2.98V50.263c0-4.381-3.587-7.939-7.989-7.939h-19.97c-4.402 0-7.988 3.558-7.988 7.939V65.14a2.99 2.99 0 002.996 2.98 2.99 2.99 0 002.996-2.98V50.263c0-1.09.89-1.98 1.996-1.98 1.106 0 1.997.89 1.997 1.98V95.06a2.99 2.99 0 002.996 2.98z"
            />
          </RNSVGGroup>
          <RNSVGCircle
            cx={40.5}
            cy={40}
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            r={38.5}
            stroke={4287274946}
            strokeWidth={3}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333239",
              "fontFamily": "Inter-Bold",
              "fontSize": 13,
              "lineHeight": 22,
            }
          }
        >
          0xd10199414d...
        </Text>
        <Text
          style={
            Object {
              "color": "#73839D",
              "fontFamily": "Inter-Regular",
              "fontSize": 12,
              "lineHeight": 20,
            }
          }
        >
          Manager since Ago 24, 2021
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
